bcipy/simulator/demo/button_press_data_processor.py

```diff
@@ -0,0 +1,139 @@
+"""Process raw button press data."""
+
+from typing import List, Tuple
+
+import numpy as np
+
+from bcipy.acquisition import devices
+from bcipy.acquisition.devices import DeviceSpec
+from bcipy.acquisition.multimodal import ContentType
+from bcipy.core.list import grouper
+from bcipy.core.parameters import Parameters
+from bcipy.core.raw_data import RawData
+from bcipy.core.triggers import TriggerType
+from bcipy.display.main import ButtonPressMode, PreviewParams
+from bcipy.io.load import load_raw_data
+from bcipy.simulator.data.data_process import (DecodedTriggers,
+                                               ExtractedExperimentData,
+                                               RawDataProcessor, ReshapedData,
+                                               TimingParams)
+from bcipy.simulator.demo.button_press_utils import (should_press_button,
+                                                     simulate_raw_data,
+                                                     switch_device)
+from bcipy.simulator.exceptions import IncompatibleParameters
+from bcipy.task.data import EvidenceType
+
+
+class ButtonPressDataProcessor(RawDataProcessor):
```

Comment:
This could perhaps be split into a generator and processor. We do have button press data that could be loaded or mocked.

Reply:
As well, it can be moved to the existing data_process module